datafusion/physical-expr/src/window/window_expr.rs

```diff
@@ -186,6 +186,10 @@ pub trait AggregateWindowExpr: WindowExpr {
         accumulator: &mut Box<dyn Accumulator>,
     ) -> Result<ScalarValue>;
 
+    /// Indicates whether this window function always produces the same result
+    /// for all rows in the partition.
+    fn is_constant(&self) -> bool;
```

Comment:
```suggestion
    fn is_constant_in_partition(&self) -> bool;
```
nit: It would be great to be more specific.

Reply:
Good point, I'll address it a bit later.